gems/smithy/lib/smithy/views/client/types.rb

```diff
@@ -88,9 +83,42 @@ def attribute_docstrings
             docstrings.each do |docstring|
               lines << "  #{docstring}"
             end
-            lines << "  @return [#{Model::YARD.type(@model, @id, @target)}]"
+            lines << "  @return [#{@doc_type}]"
             lines
           end
+
+          def default?
+            traits = @member.fetch('traits', {})
+            traits.key?('smithy.api#default') && !traits.key?('smithy.api#clientOptional')
+          end
+
+          def default
+            default = @member.dig('traits', 'smithy.api#default')
+            case @target['type']
+            when 'blob' then "Base64.strict_decode64('#{default}')"
+            when 'bigDecimal' then "BigDecimal('#{default}')"
+            when 'document' then document(default)
+            when 'enum', 'string' then "'#{default}'"
+            when 'timestamp' then timestamp(default)
+            else default
+            end
+          end
+
+          def document(default)
+            case default
+            when nil then 'nil'
```

Comment:
Should a check for nil default be done above and default just be skipped completely if its nil or is this a special document case?

Reply:
This is a document case.